app/controller/button/SpatialQueryButton.js

```diff
@@ -128,7 +147,7 @@ Ext.define('CpsiMapview.controller.button.SpatialQueryButtonController', {
     onWfsExecuteSuccess: function (response) {
         var me = this;
         var view = me.getView();
-        var mapComp = CpsiMapview.view.main.Map.guess();
+        var mapComp = me.mapComponent || BasiGX.util.Map.getMapComponent();
```

Comment:
In #116, @chrismayer asked me to use `CpsiMapview.view.main.Map.guess();` in favor of `BasiGX.util.Map.getMapComponent();` since this detects the cpsi map-panel directly by it's `xtype`. Maybe the guess function should be enhanced to allow a "fuzzy" search if no `cmv_map` was found. What do you think @geographika @chrismayer  ?

Reply:
Ah ok I thought that function was searching for a map rather than mapComponent. I was running with the latest cpsi-mapview project and getting an error when loading the application - you didn't encounter this?

Reply:
I think the clear way is to inject the map component when creating SpatialQueryButton component. The guess methods should only be used as fallback.
Nevertheless the guess function of BasiGX seems more flexible and thus it is maybe better to use it here.

Reply:
> Ah ok I thought that function was searching for a map rather than mapComponent. I was running with the latest cpsi-mapview project and getting an error when loading the application - you didn't encounter this?

@geographika : I successfully checked the tool in cpsi-mapview as well, but maybe in an older version than you. However, could you please so kind to include the suggestion of @chrismayer of passing `map` / `mapComponent` to `SpatialQueryButton`?